polyA/_runners.py

```diff
@@ -0,0 +1,526 @@
+            repeat_scores,
+            align_matrix,
+        )
+
+        # check if TR columns were added after last alignment
+        # TR cols before alignments were accounted for in PadSeqs
+        max_col_index: int = max(repeat_scores)
+        if max_col_index + 1 > cols:
+            cols = max_col_index + 1
+
+        # add TR cols to NonEmptyColumns
+        for tr_col in repeat_scores:
+            col_set = set(non_empty_columns)  # only add new columns
+            col_set.add(tr_col)
+            non_empty_columns = list(col_set)
+        non_empty_columns.sort()
+
+    else:
+        confidence_matrix = fill_confidence_matrix(
+            non_empty_columns_trailing,
+            subfam_counts,
+            subfamily_matrix,
+            active_cells_trailing,
+            align_matrix,
+        )
+
+    # add skip state to consensus matrix
+    # wait till after incase NonEmptyColumns is updated by TR stuff
+    for j in non_empty_columns:
+        consensus_matrix[0, j] = 0
+
+    support_matrix = fill_support_matrix(
+        rows,
+        chunk_size,
+        start_all,
+        non_empty_columns,
+        starts_matrix,
+        stops_matrix,
+        subfamily_matrix,
+        confidence_matrix,
+        consensus_matrix,
+    )
+
+    collapsed_matrices = collapse_matrices(
+        rows,
+        non_empty_columns,
+        subfamily_matrix,
+        strands_matrix,
+        active_cells,
+        support_matrix,
+        consensus_matrix,
+    )
+
+    support_matrix_collapse = collapsed_matrices.support_matrix
+    subfams_collapse = collapsed_matrices.subfamilies
+    active_cells_collapse = collapsed_matrices.active_rows
+    consensus_matrix_collapse = collapsed_matrices.consensus_matrix
+    strand_matrix_collapse = collapsed_matrices.strand_matrix
+    subfams_collapse_index = collapsed_matrices.subfamily_indices
+    rows = collapsed_matrices.row_num_update
+
+    if repeat_scores is not None:
+        # give different TRs consensus positions that don't allow them to be stitched
+        tr_consensus_pos = 1000000
+        prev_tr_col = 0
+        for tr_col in repeat_scores:
+            if prev_tr_col != tr_col - 1:
+                tr_consensus_pos -= 500
+            consensus_matrix_collapse[rows - 1, tr_col + 1] = tr_consensus_pos
+            prev_tr_col = tr_col
+
+    # if command line option included to output support matrix for heatmap
+    if outfile_heatmap:
+        print_matrix_support(
+            cols,
+            start_all,
+            target.chrom_start,
+            support_matrix_collapse,
+            subfams_collapse,
+            outfile=outfile_heatmap,
+        )
+
+    (
+        ProbMatrixLastColumn,
+        OriginMatrix,
+        SameSubfamChangeMatrix,
+    ) = fill_probability_matrix(
+        same_prob_skip,
+        SAME_PROB_LOG,
+        change_prob_log,
+        change_prob_skip,
+        non_empty_columns,
+        collapsed_matrices,
+    )
+
+    node_id = START_ID
```

Comment:
This start value must result in a totally unique set of IDs for this shard relative to all other shards